redis.conf

```diff
@@ -823,6 +823,18 @@ replica-priority 100
 # replica-announce-ip 5.5.5.5
 # replica-announce-port 1234
 
+# You can configure a replica instance to redirect data access commands
+# to its master or not, excluding commands such as:
+# INFO, AUTH, ROLE, SUBSCRIBE, UNSUBSCRIBE, PUBLISH.
+#
+# When enabled, a replica instance will reply "-MOVED -1 master-ip:port"
+# for data access commands. Normally these are write or read commands, if
+# you want to run read commands while replica-enable-redirect is enabled,
+# use the READONLY command first.
+#
+# This config only takes effect in standalone mode.
```

Comment:
```suggestion
# This config only takes effect in non-cluster mode.
```

i'll start a separate thread on terminology.
i've seen the discussion above suggesting that every collection of redis instances (master + set of replicas) is called a cluster, and i'm against that, but i'm also against using the term "standalone".

the best thing i can think of is "cluster mode enabled" / "redis cluster" (or in short "cluster" or CME" which i think are already heavily used)
vs "cluster mode disabled", or "non-cluster" (i'm not a fun of "cluster mode disabled", and certainly not "CMD")

Reply:
Why not "standalone mode"? It's used in `INFO` (`redis_mode:standalone` since 8246e58abe, 2012), `HELLO` (`"mode" => "standalone"`) and the splash screen.

The modes are "sentinel", "cluster" and "sentinel". AFAICT that's the original terminology.

![spash](https://github.com/redis/redis/assets/273886/ba68623c-ab55-4fa4-b2ae-3b02332c0dc9)

Reply:
ok, good point.
i don't like it, but let's stick with it.

Reply:
I don't think we are beholden to a decision made 10 years ago by someone who no longer works on the project. The terminology is barely used in the codebase, and really doesn't make much sense. I'm going to continue to non-cluster mode or cluster mode disabled.

Reply:
These words make me feel uncomfortable. You didn't give enough respect to antirez. As the creator of Redis, his contributions cannot be ignored. Moreover, in the future, each of us may leave this project, but the code and spirit will remain.

"Standalone" is a conceptual description. The reason why it has not been applied to Redis code and client code is that Redis has been in standalone mode since its inception. The cluster mode was invented later and needed to be explicitly written in the code. This is not a difference between backwardness and advancement, but simply a difference in time. Because cluster mode appeared later, the word "cluster" needs to be used in Redis and clients code to indicate its difference from standalone mode.

Reply:
it's not about respect, a decision that was made long ago, and possibly without investing much thought into it, could have become outdated over time.
The fact is that both myself and Madelyn weren't even aware of the few places where "standalone" was mentioned (even though washing our eyes on these prints, and even source code lines many many times).
IMHO, the history of how we got here is irrelevant, what's relevant is that this term isn't very intuitive (vs "non-cluster"), and although it is admittedly used in several places, it's not really widely used, so if we have a better one, we can afford to break it.

I'll give another example: till redis 7.0, the words "script", "Lua", and "EVAL" were all used interchangeably (for instance in INFO and CONFIG). now hat we had several different "scripting" commands, and even for the same command, we have several different programming languages, we had to clean this up.
The point is, that with time we sometime have to re-think past decisions and terminology, in some even if it causes interface changes, and it has nothing to do with respecting the history.

Reply:
I don't object to the terminology "non-cluster". What I want to emphasize is that standalone(non-cluster) and cluster are two very different modes. They are very different for Redis itself, its client, and especially for user business. When using cluster, users must plan their keyname settings so that multi-key operations can be performed, such as using hashtags. Therefore, for now, I don't want to mix standalone(non-cluster) and cluster, because we haven't solved many issues such as cross-slot, maintenance operations such as failover and building replication, and data storage.

Reply:
> Therefore, for now, I don't want to mix standalone(non-cluster) and cluster, because we haven't solved many issues such as cross-slot, maintenance operations such as failover and building replication, and data storage.

Are we all not directionally aligned we want to do this though? I'm under the impression we want to move all configurations to a mode where the clusterbus is running and it supports operations like `CLUSTER FAILOVER`. I don't think this needs to necessarily solve the sharding problem, since you could have an unsharded database.

Reply:
I think you misunderstood me. As of now, I am not inclined towards an unsharded-cluster. In my previous response, I was only pointing out its issues, not indicating that I support implementation it.

Reply:
> I think you misunderstood me. As of now, I am not inclined towards an unsharded-cluster. In my previous response, I was only pointing out its issues, not indicating that I support implementation it now.

I understand you aren't for it right now. I'm asking if you are inclined towards an unsharded-cluster eventually. My thinking was that we were going to target a lot of cluster improvements for Redis 8, including supporting unsharded-clusters. (I think you indicated some doubt in another comment, and that's probably warranted at this point. We haven't made much progress there).